LambdaEngine/Include/ECS/ComponentArray.h

```diff
@@ -66,15 +85,24 @@ namespace LambdaEngine
 	{
 		auto indexItr = m_EntityToIndex.find(entity);
 		VALIDATE_MSG(indexItr != m_EntityToIndex.end(), "Trying to get a component that does not exist!");
-		return m_Data[indexItr->second];
+
+		Comp& component = m_Data[indexItr->second];
+
+		if constexpr (Comp::HasDirtyFlag())
+		{
+			component.Dirty = true;
+		}
+
+		return component;
 	}
 
 	template<typename Comp>
-	inline void ComponentArray<Comp>::DeleteEntity(Entity entity)
+	inline const Comp& ComponentArray<Comp>::GetData(Entity entity) const
```

Comment:
You have inline in cpp but not in header. How come?

Reply:
This is the header file